### DISCUSSION TITLE
MAILER_DSN invalid

### DISCUSSION THREAD

**Discussion by giosal** [symfony/symfony#57509] [Q&A] [answered]
### Symfony version(s) affected

6.4.8

### Description

When using a password containing "?" in the MAILER_DSN string in .env file, the Symfony is not able to parse the string correctly thinking that it's a special character.
For example:

```env
MAILER_DSN=smtp://user@domain.org:BVVkJ,Ut?GN4@mail.domain.org:587?encryption=tls
```

(username and domain replaced for privacy)

Is there an option to use the passwords containing DSN special characters, such as "?", "@", ":" etc. ?

### How to reproduce

Put a password containing one of the special characters in the MAILER_DSN string in .env file

**Comment by derrabus** [comment, marked as answer]
```env
MAILER_DSN=smtp://user%40domain.org:BVVkJ%2CUt%3FGN4@mail.domain.org:587?encryption=tls
```

You can use [PHP's `rawurlencode()` function](https://www.php.net/rawurlencode) to escape special characters like `@` or `?`, so you can safely use them in URLs.

**Reply by giosal** [reply]
Great, thank you!

**Reply by apoorva-shah** [reply]
Hello @derrabus 

What to do for password like? s+z+afg
Its encong will be s%2Bz%2Bafg

Not working.  I tried one more % as prefix. Not working. Can you please suggest?
